### ISSUE TITLE
[5.8.0] Type error in `toBooleanOrNull`: Expected `NO_VALUE` to be a `RelationshipValue`

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13194]
When running the following query:
```cypher
RETURN toBooleanOrNull(startNode(null))
```

I encountered the following error:
```
Neo4jError: Neo.ClientError.Statement.TypeError (Invalid input for function 'startNode()': Expected NO_VALUE to be a RelationshipValue)
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.8.0 enterprise version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following queries and observe it throws an error:
```cypher
RETURN toBooleanOrNull(startNode(null))
```

### Expected behavior
The query should run successfully

### Actual behavior
**5.7.0**: The query runs successfully
**5.8.0**: The query fails with the error message `Neo4jError: Neo.ClientError.Statement.TypeError (Invalid input for function 'startNode()': Expected NO_VALUE to be a RelationshipValue)`

**Comment by arnefischereit** [comment]
Hi @DominicWuest 

Thank you for reporting another bug in this area. I could reproduce the issue. We will further investigate and come back to you with our findings.

Regards,

Arne

---

[Internal link for reference](https://trello.com/c/tCc0yxxi)

**Comment by Lojjs** [comment]
@DominicWuest This change in behaviour was introduced by mistake in a refactoring. It has now been restored to the behaviour it had in 5.7. The fix will be included in Neo4j 5.10
